databuilder/databuilder/extractor/bigquery_usage_extractor.py

```diff
@@ -99,29 +109,54 @@ def _create_records(self, refResources: List[dict], resourcesProcessed: int, ema
             return
 
         for refResource in refResources:
-            key = TableColumnUsageTuple(database='bigquery',
-                                        cluster=refResource['projectId'],
-                                        schema=refResource['datasetId'],
-                                        table=refResource['tableId'],
-                                        column='*',
-                                        email=email)
+            tableId = refResource['tableId']
+            datasetId = refResource['datasetId']
+
+            if self._is_anonymous_dataset(datasetId) or self._is_wildcard_table(tableId):
+                continue
+
+            tableId = self._remove_table_decorators(tableId)
+
+            if self._is_sharded_table(tableId):
+                tableId = tableId[:-BigQueryTableUsageExtractor.DATE_LENGTH]
+
+            if self.count_tables_only_from_project_id:
+                if refResource['projectId'] == self.project_id:
+                    key = TableColumnUsageTuple(database='bigquery',
+                                                cluster=refResource['projectId'],
+                                                schema=datasetId,
+                                                table=tableId,
+                                                column='*',
+                                                email=email)
+                else:
+                    LOGGER.debug(f'Not counting usage for {refResource} since '
+                                 f'{BigQueryTableUsageExtractor.COUNT_TABLES_ONLY_FROM_PROJECT_ID_KEY} is True')
+                    continue
+            else:
+                key = TableColumnUsageTuple(database='bigquery',
```

Comment:
I wonder for other non rdbms store (graph db), we should also ignore the tables that are not from the same project?

Reply:
Yes, we could ignore usage counts for tables from other projects even in the case of graph db, but made it configurable in case graph db users want to get the usage of all the referenced tables from their queries, even if they are not from the same project. 
Should I make ‘count_tables_only_from_project_Id_key’ the default behavior?

Reply:
yeap, I think `count_tables_only_from_project_Id_key` as default makes sense

Reply:
Updated the PR!